networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/reputation/ReputationManager.java

```diff
@@ -13,156 +13,32 @@
 
 package tech.pegasys.teku.networking.p2p.reputation;
 
-import com.google.common.base.MoreObjects;
-import java.util.EnumSet;
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-import org.hyperledger.besu.plugin.services.MetricsSystem;
-import tech.pegasys.teku.infrastructure.collections.cache.Cache;
-import tech.pegasys.teku.infrastructure.collections.cache.LRUCache;
-import tech.pegasys.teku.infrastructure.metrics.TekuMetricCategory;
-import tech.pegasys.teku.infrastructure.time.TimeProvider;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.networking.p2p.network.PeerAddress;
 import tech.pegasys.teku.networking.p2p.peer.DisconnectReason;
-import tech.pegasys.teku.networking.p2p.peer.NodeId;
 
-public class ReputationManager {
+public interface ReputationManager {
   // This is not a big ban, we expect the peer could be usable very soon
-  static final UInt64 COOLDOWN_PERIOD = UInt64.valueOf(TimeUnit.MINUTES.toSeconds(2));
+  UInt64 COOLDOWN_PERIOD = UInt64.valueOf(TimeUnit.MINUTES.toSeconds(2));
   // It's a big ban, we expect that the peer is not useful until major changes, for example,
   // software update
-  static final UInt64 BAN_PERIOD = UInt64.valueOf(TimeUnit.HOURS.toSeconds(12));
+  UInt64 BAN_PERIOD = UInt64.valueOf(TimeUnit.HOURS.toSeconds(12));
 
-  static final int LARGE_CHANGE = 10;
-  static final int SMALL_CHANGE = 3;
-  private static final int DISCONNECT_THRESHOLD = -LARGE_CHANGE;
-  private static final int MAX_REPUTATION_SCORE = 2 * LARGE_CHANGE;
+  int LARGE_CHANGE = 10;
+  int SMALL_CHANGE = 3;
```

Comment:
Do I need to make it so that these static values can change with each implementation?

Reply:
I would just move all of these values to DefaultReputationManager same as before, since they seem more like implementation details.

Reply:
I would just move all static finals back to the DefaultReputationManager instead of the interface since they look like implementation details.